src/parser.rs

```diff
@@ -4227,6 +4230,33 @@ impl<'a> Parser<'a> {
         })
     }
 
+    pub fn parse_tablesample_args(&mut self) -> Result<Option<TableSample>, ParserError> {
```

Comment:
There are subtle differences in how tablesample is parsed in postgresql vs other dialects. Eg: mssql expects the sampling_fraction to also specify `rows/percent`. Does it make sense to have this check (and tests) for postgresql only?

Reply:
Would it make sense to add `parse_tablesample_args` to the `Dialect` trait and have a default implementation that returns `None`? The dialects that want to support it can add their custom parsing. The `TableSample` struct would still need to be a superset of all supported options across all dialects though.

Reply:
Isn't it better to use the dialect locally to handle impl differences, instead of adding it to the `Dialect` trait?

Otherwise the parser would end up with incompatible concepts around the code: some places use the dialect to validate, others use the `Dialect` trait to implement specific concepts. Would be nice to have a standard around where to put specific implementations

Reply:
Isn't it better to use the dialect locally to handle impl differences, instead of adding it to the `Dialect` trait?

Otherwise the parser would end up with incompatible concepts around the code: some places use the dialect to validate, others use the `Dialect` trait to implement specific concepts. Would be nice to have a standard around where to put specific implementations